### ISSUE TITLE
Firebase is much slower to use with SPM over CocoaPods

### ISSUE THREAD

**Issue by PomegranateApps** [apple/swift-package-manager#5545]
### Expected behavior
I would expect an installation of Firebase with SPM or CocoaPods to be similar.

### Actual behavior
There is a significant delay every time I launch my app with Firebase using SPM over CocoaPods. There's virtually no difference when using CocoaPods, but with SPM it takes 1-2 seconds extra before my app with launch on the simulator or device.

### Swift & OS version (output of `swift --version && uname -a`)
Darwin Kernel Version 21.4.0
root:xnu-8020.101.4~15/RELEASE_X86_64 x86_64

**Comment by sena-mike** [comment]
I noticing package resolving in Xcode takes several seconds as well even after caching.

**Comment by tomerd** [comment]
@PomegranateApps iiuc you are talking about compile time, while @sena-mike iuuc you are talking about resolving dependencies. is my understanding correct and these are two separate topics?

cc @neonichu @abertelrud

**Comment by neonichu** [comment]
Runtime behavior of apps which use a library seems more of a question for the library author itself since I don't think we're in a position to really decide where such an issue lies.

**Comment by tomerd** [comment]
> Runtime behavior of apps which use a library seems more of a question for the library author itself since I don't think we're in a position to really decide where such an issue lies.

correct. I wonder tho if there is some difference in static / dynamic linking that could impact this?

**Comment by tomerd** [comment]
@PomegranateApps can you please refine where you see the delay: is it the build in Xcode prior to launching the app in the simulator, or is this the app launch time in the simulator?
